File: solutions/python/977.py
# Detail：
# Runtime: 511 ms
# Memory Usage: 16.2 MB
class Solution:
    def sortedSquares(self, nums: List[int]) -> List[int]:
        n = len(nums)
        i,j,k = 0,n - 1,n - 1
        ans = [-1] * n
        while k >= 0:
            left = nums[i] ** 2
            right = nums[j] ** 2
            if left < right:
                ans[k] = right
                j -= 1
            else:
                ans[k] = left
                i += 1
            k -= 1
        return ans

File: solutions/c++/977.cpp
// Details:
// Runtime: 43 ms, faster than 72.39% of C++ online submissions for Squares of a Sorted Array.
// Memory Usage: 25.9 MB, less than 81.50% of C++ online submissions for Squares of a Sorted Array.
class Solution {
public:
    vector<int> sortedSquares(vector<int>& nums) {
        int k = nums.size() - 1;
        vector<int> results(nums.size(), 0);
        for (int i = 0, j = nums.size() - 1; i <= j; ){
            if (nums[i] * nums[i] < nums[j] * nums[j]){
                results[k--] = nums[j] * nums[j];
                j--;
            } else{
                results[k--] = nums[i] * nums[i];
                i++;
            }
        }
        return results;
    }
};